app/controllers/catalog_controller.rb

```diff
@@ -406,12 +406,17 @@ def x_show
       else
         @record = ServiceTemplateCatalog.find(params[:id])
       end
+    elsif x_active_tree == :sandt_tree
+      assert_privileges("catalog_items_view")
+
+      identify_catalog(params[:id])
+      @record ||= ServiceTemplateCatalog.find(params[:id])
     elsif x_active_tree == :ot_tree
       assert_privileges("orchestration_templates_view")
 
       @record ||= OrchestrationTemplate.find(params[:id])
     else
-      assert_privileges("st_catalog_view")
+      assert_privileges("svc_catalog_provision", "svc_catalog_archive", "svc_catalog_unarchive")
```

Comment:
Since the Service Catalogs don't have a permission for view and only have modify:
<img width="314" alt="Screenshot 2023-09-26 at 2 58 11 PM" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/32444791/6c0bce4a-4eef-49ba-a204-7c54f8b1efb2">

I added all three permissions in the assert_privileges check where if any of them are true it will successfully render the Service Catalog summary page.

Reply:
If we create a role without the Order Services feature checked but Archive and/or Unarchive Services is checked we can still see the Service Catalogs summary page when clicking on the table but when we click the order button we are presented with the correct error that this user lacks the required permissions:
<img width="1377" alt="Screenshot 2023-09-26 at 3 03 24 PM" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/32444791/ffd729cd-c3bd-4b59-8afb-f2e376755c46">
<img width="1374" alt="Screenshot 2023-09-26 at 3 03 44 PM" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/32444791/028001aa-7a4a-4848-82e0-b2ed1e6f9b29">

Reply:
That's great @GilbertCherrie - in a follow up we should consider adding a view permission and then we don't have to "guess".  Instead we can use your guessing logic in a database migration to add the view role to existing roles only once.

Reply:
@Fryguy We could add a view permission but it would require other changes. The other catalog options have a view and modify permission that are separate because they have a checkbox table which allows you to modify them without being able to view their summary pages. The service catalogs don't have the checkbox tables so you would run into the weird situation where if you have a role that has modify permissions for service catalogs but no view permissions than it makes it the modify permissions useless since you can't modify them without viewing (as seen with the order button on the summary page above).
If we do choose to add a view permission to service catalogs then the table would need to be changed to have checkboxes so users can modify service catalogs without viewing. I'm assuming it was intentionally created like this because you can only modify 1 service catalog at a time which you wouldn't need checkboxes for. Let me know what you think.